File: exchange-desktop-hmekhatib-main/src/main/java/com/hmekhatib/exchange/api/model/Post.java
package com.hmekhatib.exchange.api.model;

import com.google.gson.annotations.SerializedName;

public class Post {
    @SerializedName("added_date")
    public String addedDate;
    @SerializedName("id")
    public Integer id;
    @SerializedName("user_id")
    public Integer userid;
    @SerializedName("lbp_amount")
    public Float lbpAmount;
    @SerializedName("usd_amount")
    public Float usdAmount;
    @SerializedName("typeSell")
    public Boolean typeSell;
    @SerializedName("username")
    public String username;
    @SerializedName("postid")
    public Integer postid;
    @SerializedName("trade")
    public String trade;

    public String getTrade() {
        return trade;
    }

    public String getAddedDate() {
        return addedDate;
    }

    public Integer getId() {
        return id;
    }

    public Integer getUserid() {
        return userid;
    }

    public Float getLbpAmount() {
        return lbpAmount;
    }

    public Float getUsdAmount() {
        return usdAmount;
    }

    public Boolean getTypeSell() {
        return typeSell;
    }


    public String getUsername() {
        return username;
    }

    public Integer getPostid() {
        return postid;
    }

    public Post(Float usdAmount, Float lbpAmount, Boolean typeSell){
        this.lbpAmount = lbpAmount;
        this.usdAmount = usdAmount;
        this.typeSell= typeSell;
    }

    public Post(Integer postid){
        this.postid = postid;
    }
}

File: exchange-android/exchange-android-hmekhatib/app/src/main/java/com/hmekhatib/currencyexchange/api/model/Post.kt
package com.hmekhatib.currencyexchange.api.model

import com.google.gson.annotations.SerializedName
import java.util.*

class Post {
    @SerializedName("added_date")
    var addedDate: String? = null
    @SerializedName("id")
    var id: Int? = null
    @SerializedName("user_id")
    var userid: Int? = null
    @SerializedName("usd_amount")
    var usdAmount: Float? = null
    @SerializedName("lbp_amount")
    var lbpAmount: Float? = null
    @SerializedName("typeSell")
    var usdToLbp: Boolean? = null
    @SerializedName("username")
    var username: String? = null
    @SerializedName("postid")
    var postID: Int? = null

}